src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs

```diff
@@ -71,5 +72,42 @@ public static void AssertRoundtrips<T>(T expected, params Func<T, object>[] addi
                 }
             }
         }
+
+        public static void AssertExceptionDeserializationFails<T>() where T : Exception
+        {
+            // .NET Core throws a PlatformNotSupportedException for deserializing many exceptions.
+            if (!PlatformDetection.IsNetCore)
```

Comment:
Aren't we going to have similar errors for .NET Native?  Should there be a TODO here to incorporate it once that functionality is avaialble (if it's not already)?